pkgs/by-name/fo/forecast/package.nix

```diff
@@ -0,0 +1,107 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+
+  # nativeBuildInputs
+  just,
+  pkg-config,
+  wrapGAppsHook3,
+
+  # buildInputs
+  libxkbcommon,
+  openssl,
+  vulkan-loader,
+  wayland,
+  nix-update-script,
+
+  cosmic-icons,
+}:
+let
+  commitDate = "2024-09-06";
+in
+rustPlatform.buildRustPackage rec {
+  pname = "forecast";
+  version = "0-unstable-${commitDate}";
+
+  src = fetchFromGitHub {
+    owner = "cosmic-utils";
+    repo = "forecast";
+    rev = "75e6cf93348c928c0f5b20e45b619f7c9a44f9c9";
+    hash = "sha256-//wVcK18b+uykoDACnUrnRdn++91Umgi877GxwcCwUQ=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "accesskit-0.12.2" = "sha256-1UwgRyUe0PQrZrpS7574oNLi13fg5HpgILtZGW6JNtQ=";
+      "atomicwrites-0.4.2" = "sha256-QZSuGPrJXh+svMeFWqAXoqZQxLq/WfIiamqvjJNVhxA=";
+      "clipboard_macos-0.1.0" = "sha256-cG5vnkiyDlQnbEfV2sPbmBYKv1hd3pjJrymfZb8ziKk=";
+      "cosmic-config-0.1.0" = "sha256-zamYPvxmIqh4IT4G+aqceP1mXNNBA1TAcJwAtjlbYAU=";
+      "cosmic-text-0.12.1" = "sha256-5Gk220HTiHuxDvyqwz1Dwr+BaLvH/6X7M14IirQzcsE=";
+      "d3d12-0.19.0" = "sha256-usrxQXWLGJDjmIdw1LBXtBvX+CchZDvE8fHC0LjvhD4=";
+      "glyphon-0.5.0" = "sha256-j1HrbEpUBqazWqNfJhpyjWuxYAxkvbXzRKeSouUoPWg=";
+      "smithay-clipboard-0.8.0" = "sha256-4InFXm0ahrqFrtNLeqIuE3yeOpxKZJZx+Bc0yQDtv34=";
+      "softbuffer-0.4.1" = "sha256-a0bUFz6O8CWRweNt/OxTvflnPYwO5nm6vsyc/WcXyNg=";
+      "taffy-0.3.11" = "sha256-SCx9GEIJjWdoNVyq+RZAGn0N71qraKZxf9ZWhvyzLaI=";
+      "winit-0.29.10" = "sha256-ScTII2AzK3SC8MVeASZ9jhVWsEaGrSQ2BnApTxgfxK4=";
+    };
+  };
+
+  nativeBuildInputs = [
+    just
+    pkg-config
+    wrapGAppsHook3
+  ];
+
+  dontUseJustBuild = true;
+  dontUseJustCheck = true;
+
+  buildInputs = [
+    libxkbcommon
+    openssl
+    vulkan-loader
+    wayland
+  ];
+
+  env = {
+    # COSMIC applications now uses vergen for the About page
+    # Update the COMMIT_DATE to match when the commit was made
+    VERGEN_GIT_COMMIT_DATE = commitDate;
+    VERGEN_GIT_SHA = src.rev;
+  };
+
+  postFixup = ''
+    wrapProgram $out/bin/${meta.mainProgram} \
+      --suffix XDG_DATA_DIRS : "${cosmic-icons}/share" \
+      --prefix LD_LIBRARY_PATH : "${
+        lib.makeLibraryPath [
+          libxkbcommon
+          wayland
+        ]
+      }"
+  '';
+
+  justFlags = [
+    "--set"
+    "prefix"
+    (placeholder "out")
+    "--set"
+    "bin-src"
+    "target/${stdenv.hostPlatform.rust.cargoShortTarget}/release/${meta.mainProgram}"
```

Comment:
This should be written literall

Reply:
Substituting `stdenv.hostPlatform.rust.cargoShortTarget` you mean ?
It depends on the architecture so we cannot hardcode a string here I guess.

Reply:
Yeah totally, I more meant meta.mainProgram

Reply:
Ok, then I have removed them.